templates/terraform/resource.erb

```diff
@@ -189,20 +189,28 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
     }
 <%    end -%>
 <%  end -%>
+    billingProject := ""
+
+    billingProject, _ = getBillingProject(d, config)
+    if billingProject != "" {
+        config.UserProjectOverride = true
```

Comment:
We can skip doing this- we can consider the configuration of `config.UserProjectOverride` as separate from whether a `billing_project` is configured or not.

I guess there's not really a scenario where having `billing_project` set and `user_project_override` unset/false makes sense, but I'd rather we allow that unintuitive configuration than override the user's setting.

Reply:
ACM needs UserProjectOverride to be set to true so the header is sent. Is setting the header on every Google API call a problem?

Reply:
Yep- accounts need to be granted a specific permission (`serviceusage.services.use`) on a project to be allowed to use it with the header. See https://www.terraform.io/docs/providers/google/guides/provider_reference.html#user_project_override-1.

Reply:
It is better now, i added a product/resource override so if that key is present then the header is set.